File: test_api.dart
import 'package:dio/dio.dart';

void main() async {
  final dio = Dio();
  final headers = {
    'x-rapidapi-key': '7f02dcb732mshe0adcde39d9a2b1p183a68jsn1c4b7d554208',
    'x-rapidapi-host': 'download-videos-tiktok.p.rapidapi.com',
  };
  const testUrl =
      'https://www.tiktok.com/@shaziofficial__7/video/7488057209352703278';

  try {
    print('Testing GET /');
    final res1 = await dio.get(
      'https://download-videos-tiktok.p.rapidapi.com/',
      queryParameters: {'url': testUrl},
      options: Options(headers: headers),
    );
    print('GET / Success: \n${res1.data}');
  } catch (e) {
    print('GET / Error: $e');
  }

  try {
    print('\nTesting POST /');
    final res2 = await dio.post(
      'https://download-videos-tiktok.p.rapidapi.com/',
      data: {'url': testUrl, 'hd': '1'},
      options: Options(headers: headers),
    );
    print('POST / Success: \n${res2.data}');
  } catch (e) {
    print('POST / Error: $e');
  }

  try {
    print('\nTesting GET /video');
    final res4 = await dio.get(
      'https://download-videos-tiktok.p.rapidapi.com/video',
      queryParameters: {'url': testUrl},
      options: Options(headers: headers),
    );
    print('GET /video Success: \n${res4.data}');
  } catch (e) {
    print('GET /video Error: $e');
  }

  try {
    print('\nTesting GET /download');
    final res5 = await dio.get(
      'https://download-videos-tiktok.p.rapidapi.com/download',
      queryParameters: {'url': testUrl},
      options: Options(headers: headers),
    );
    print('GET /download Success: \n${res5.data}');
  } catch (e) {
    print('GET /download Error: $e');
  }
}

File: test_api.py
import urllib.request
import json
import urllib.error

url = "https://download-videos-tiktok.p.rapidapi.com/"
headers = {
    "x-rapidapi-host": "download-videos-tiktok.p.rapidapi.com",
    "x-rapidapi-key": "7f02dcb732mshe0adcde39d9a2b1p183a68jsn1c4b7d554208",
    "Content-Type": "application/json"
}

def test_endpoint(endpoint, method="GET", data=None):
    req_url = url + endpoint if not url.endswith('/') and not endpoint.startswith('/') else url.rstrip('/') + '/' + endpoint.lstrip('/')
    req = urllib.request.Request(req_url, headers=headers, method=method)
    if data:
        req.data = json.dumps(data).encode('utf-8')
    try:
        with urllib.request.urlopen(req) as response:
            print(f"[{method}] /{endpoint} -> {response.status}")
            print(response.read().decode('utf-8')[:200])
    except urllib.error.HTTPError as e:
        print(f"[{method}] /{endpoint} -> Error: {e.code}")
    except Exception as e:
        print(f"[{method}] /{endpoint} -> Error: {e}")

endpoints = ["download", "info", "tiktok", "video", "tiktok/info", "tiktok/video", "tiktok/download", "api/download", "api/video", "aweme/v1/web/aweme/detail"]
for ep in endpoints:
    test_endpoint(ep, method="GET")
    test_endpoint(ep, method="POST", data={"url": "https://www.tiktok.com/@tiktok/video/7106594312292453675"})
